core/capabilities/syncer.go

```diff
@@ -163,21 +175,86 @@ func (s *registrySyncer) Name() string {
 	return "RegistrySyncer"
 }
 
-type noOpTrigger struct {
-	info commoncap.CapabilityInfo
-	lggr logger.Logger
+type mockMercuryDataProducer struct {
+	trigger *triggers.MercuryTriggerService
+	wg      sync.WaitGroup
+	closeCh chan struct{}
+	lggr    logger.Logger
 }
 
-func (t *noOpTrigger) Info(_ context.Context) (commoncap.CapabilityInfo, error) {
-	return t.info, nil
+var _ services.Service = &mockMercuryDataProducer{}
+
+func NewMockMercuryDataProducer(trigger *triggers.MercuryTriggerService, lggr logger.Logger) *mockMercuryDataProducer {
+	return &mockMercuryDataProducer{
+		trigger: trigger,
+		closeCh: make(chan struct{}),
+		lggr:    lggr,
+	}
 }
 
-func (t *noOpTrigger) RegisterTrigger(_ context.Context, _ chan<- commoncap.CapabilityResponse, request commoncap.CapabilityRequest) error {
-	t.lggr.Infow("no-op trigger RegisterTrigger", "workflowID", request.Metadata.WorkflowID)
+func (m *mockMercuryDataProducer) Start(ctx context.Context) error {
+	m.wg.Add(1)
+	go m.loop()
 	return nil
 }
 
-func (t *noOpTrigger) UnregisterTrigger(_ context.Context, request commoncap.CapabilityRequest) error {
-	t.lggr.Infow("no-op trigger RegisterTrigger", "workflowID", request.Metadata.WorkflowID)
+func (m *mockMercuryDataProducer) loop() {
+	defer m.wg.Done()
+
+	sleepSec := 60
```

Comment:
Do we want something a little faster than 1 min report production for demo purposes?

Reply:
The trigger still sends events every 5 seconds based on maxFrequencyMs setting in the workflow spec. Your comment makes me think that this is incorrect and we should send events only when there are newer values. I'll fix that separately.